File: java/647-Palindromic-Substrings.java
class Solution {

  public int countSubstrings(String s) {
    if (s.length() < 2) {
      return s.length();
    }
    int result = 0;
    for (int i = 0; i < s.length(); i++) {
      // Odd Length
      int left = i, right = i;
      while (
        left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)
      ) {
        result++;
        left--;
        right++;
      }
      // Even Length
      left = i;
      right = i + 1;
      while (
        left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)
      ) {
        result++;
        left--;
        right++;
      }
    }
    return result;
  }
}

File: csharp/647-Palindromic-Substrings.cs
public class Solution {
    public int CountSubstrings(string s) {
        var count = 0;
        
        for(var i = 0; i < s.Length; i++) {
            count += getPalindromeCount(s, i, i);
            count += getPalindromeCount(s, i, i + 1);
        }
        
        return count;
    }
    
    public int getPalindromeCount(string s, int l, int r) {
        var count = 0;
        
        while(l >=0 && r < s.Length && s[l] == s[r]) {
            count++;
            l--;
            r++;
        }
        return count;
    }
}